src/qforte/ucc/spqe.py

```diff
@@ -152,8 +155,8 @@ def run(self,
         self._pool_type = max_excit_rank
 
         idx = 0
-        # determinant id : excitation operator index
-        self._excitation_dictionary = {}
+        # Given a coefficient index, what is the index of the "corresponding" pool element? Used to compute the operator to add to the ansatz in macroiterations.
+        self._coeff_idx_to_pool_idx = {}
```

Comment:
I understand why you used `coeff_idx`, but perhaps replacing it by `det_idx` or `det_int` might be clearer, especially since in this part of the code the `coeff ` variable has not been defined yet.

The way I think about it is as follows.
Let's say that we are dealing with H6/STO-6G. The HF reference determinant |111111000000> corresponds to the integer 63 (111111). Let's assume that the first operator to be appended to the ansatz is a HOMO -> LUMO double excitation. This operator is associated with the determinant |111100110000>, which corresponds to the integer 207. Thus, given the integer 207 the dictionary will return 0, namely, the first element of the pool.

What do you think?